### DISCUSSION TITLE
Why in v5 some Components the CSS has increased specificity?

### DISCUSSION THREAD

**Discussion by cant89** [ant-design/ant-design#40879] [Q&A]
For instance, in v4 the properties of the `.ant-layout-sider` where defined as below:
```
.ant-layout-sider {
   // ...
}
```
While in v5 the same properties are now as:
```
:where(.css-dev-only-do-not-override-diro6f).ant-layout .ant-layout-sider {
   // ... 
}
```
The problem here is that the specificity of this CSS rule is increased.
I am not talking about the `:where(.css-dev-only-do-not-override-diro6f)`.
I am talking about the increased specificity coming from the addition of `.ant-layout`.

The consequence is that a simple override made with styled-components, such as the following, does not work anymore:

```
const StyledSider = styled(Sider)`
   // ... my custom rules
`
```

The reason is that styled-components adds a random class name to the Sider, like the following, which now is not winning over the antd rule anymore:
```
.eUNTQd {
     // ... my custom rules
}
```

Does anyone know why this was done and what is the quicker way to solve this issue? 
I cannot go through the whole application and increase the specificity of all the overrides we have made across all the components usages. 

Thanks to anyone who can help

**Comment by arvinxx** [comment]
@cant89  maybe this ?

```ts
const StyledSider = styled(Sider)`
  & .ant-layout-sider {
   // ... my custom rules
  }
`
```

**Reply by cant89** [reply]
> @cant89 maybe this ?
> 
> ```ts
> const StyledSider = styled(Sider)`
>   & .ant-layout-sider {
>    // ... my custom rules
>   }
> `
> ```

Thanks for your answer.
I know I can do it but as I said we have hundreds of cusotmizations, i would really love to avoid to do it.

I see two options:
1. There is a way to reduce the increased specificity introcuded with v5, maybe with a configuration or something else
2. There is a way to get to know all the components for which the specificity is increased so that, hopefully, they are just a few and I can punctually go and change them one by one as you suggested

Also I wonder why this has been done, since it does not seems to me that was necessary while instead it is now creating this issue.

**Reply by arvinxx** [reply]
it seems due to refactor less to cssinjs ? https://github.com/ant-design/ant-design/commit/a74fdbc911b14f8ae77b8ceec2e72466186b4e1b#diff-c7cc0a31b7742928fa2dc6d07edfe1d514fc7f04523e4a8876932ef67537e026 

@MadCcc  @zombieJ

**Reply by Wxh16144** [reply]
I think we have a problem with the refactoring and should remove the selector weights and add hashId.

**Reply by cant89** [reply]
@Wxh16144 does it means that, if the bug is confirmed, the result will be the following?
```
:where(.css-dev-only-do-not-override-diro6f) .ant-layout-sider {
   // ... 
}
```

**Reply by arvinxx** [reply]
@cant89  it seems add a prefix by design. so you should change the code by yourself. @MadCcc  you can explain the reason for this change.

**Reply by cant89** [reply]
@MadCcc do you mind sharing your opinion? This seems to be a breaking change not mentioned anywhere in the migration guide from v4 to v5

**Reply by MadCcc** [reply]
Hi there~ Sorry for late reply.

There are some reasons for this change, and this is intended when migrate to cssinjs, just as @arvinxx said.
1. When we started to migrate, `:where` was not taken into consideration. We also worried about the adjustment of css priority brought by hash, but it seems not that breaking. And if you use `highHashPriority` with StyleProvider, the priority of css would still be raised.
2. For Layout, Layout.Sider is a sub component of Layout, and we do not want users to use it alone. So the change of css priority of some components like Layout makes sense for us.

> @Wxh16144 does it means that, if the bug is confirmed, the result will be the following?
> 
> ```
> :where(.css-dev-only-do-not-override-diro6f) .ant-layout-sider {
>    // ... 
> }
> ```

I totally agree this. But the migration is completed, most of antd components have this adjustment in style, we won't treat it as a bug. Currently you could change your code like this:
```
const StyledSider = styled(Sider)`
  .ant-layout & {
   // ... my custom rules
  }
`
```

Thanks for your support for antd!
